library/psa_crypto.c

```diff
@@ -1072,6 +1072,10 @@ psa_status_t psa_destroy_key(mbedtls_svc_key_id_t key)
     }
 
 #if defined(MBEDTLS_THREADING_C)
+    /* We cannot unlock between setting the state to PENDING_DELETION
+     * and destroying the key in storage, as otherwise another thread
+     * could load the key into a new slot and the key will not be
+     * fully destroyed. */
```

Comment:
Remark (no changes requested): I think it would be a better strategy to make key loading respect `PENDING_DELETION`, because accessing storage or secure elements can take a long time. However, it is not trivial amount of work and I feel that it is premature optimisation at this point.

Reply:
Indeed, I haven't thought of that. As far as I can tell, that can't be helped without a UUID, which is out of scope for now.